### PR TITLE
ci: add ci pipeline actions

### PR DESCRIPTION
**Pull request by itsdebs**
## Description of the change

> Added Ci

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: OneTrustConsentFilter/lib/build.gradle [hunk 1]
```diff
@@ -37,7 +37,7 @@ dependencies {
     androidTestImplementation 'androidx.test.ext:junit:1.1.4'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
     compileOnly 'com.onetrust.cmp:native-sdk:202212.1.0.0'
-    compileOnly 'com.rudderstack.android.sdk:core:1.9.0-SNAPSHOT'
+    compileOnly 'com.rudderstack.android.sdk:core:1.8.1-SNAPSHOT'
```
Comment (bardisg) [inline OneTrustConsentFilter/lib/build.gradle]: does this have to be a snapshot version? I would expect a release one

File: OneTrustConsentFilter/gradle.properties [hunk 2]
```diff
@@ -20,4 +20,19 @@ kotlin.code.style=official
...
+POM_DEVELOPER_ID=Rudderstack
+POM_NAME=Rudder-Onetrust-Consent-Interceptor-Android
+POM_DESCRIPTION=RudderStack\'s One trust Interceptor.
+GROUP=com.rudderstack.android.consentfilter
+POM_LICENCE_DIST=repo
+VERSION_NAME=0.0.0
+POM_LICENCE_NAME=The MIT License (MIT)
+POM_DEVELOPER_NAME=Rudderstack, Inc.
+VERSION_CODE=0
+POM_URL=https://github.com/rudderlabs/rudder-integration-moengage-android
```
Comment (bardisg) [inline OneTrustConsentFilter/gradle.properties]: is this correct? how is moengage relevant?

File: OneTrustConsentFilter/gradle.properties [hunk 3]
```diff
@@ -20,4 +20,19 @@ kotlin.code.style=official
...
\ No newline at end of file
+android.nonTransitiveRClass=true
+android.enableJetifier=true
+POM_LICENCE_URL=http://opensource.org/licenses/MIT
+POM_DEVELOPER_ID=Rudderstack
+POM_NAME=Rudder-Onetrust-Consent-Interceptor-Android
+POM_DESCRIPTION=RudderStack\'s One trust Interceptor.
+GROUP=com.rudderstack.android.consentfilter
+POM_LICENCE_DIST=repo
+VERSION_NAME=0.0.0
```
Comment (bardisg) [inline OneTrustConsentFilter/gradle.properties:31]: will this get updated with releashed version?

File: OneTrustConsentFilter/.github/workflows/snapshot_release.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+          restore-keys: |
+            ${{ runner.os }}-gradle-core-
+
+      - name: Publish Onetrust snapshot to sonatype
+        run: ./OneTrustConsentFilter/gradlew publishToSonatype
+        env:
+          SIGNING_KEY_ID: ${{ secrets.SIGNING_KEY_ID }}
+          SIGNING_KEY_PASSWORD: ${{ secrets.SIGNING_KEY_PASSWORD }}
+
+          ORG_GRADLE_PROJECT_sonatypeUsername: ${{ secrets.NEXUS_USERNAME }}
```
Comment (bardisg) [inline OneTrustConsentFilter/.github/workflows/snapshot_release.yml]: have we added al these secrets in the repo settings?

File: OneTrustConsentFilter/.github/workflows/slack-notify.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  deploy-tag:
+    name: Notify Slack
+    runs-on: ubuntu-latest
+    steps:
+      - name: Send message to Slack channel
+        id: slack
+        uses: slackapi/slack-github-action@v1.23.0
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+          PROJECT_NAME: rudder-integration-moengage-android
```
Comment (bardisg) [inline OneTrustConsentFilter/.github/workflows/slack-notify.yml]: project name is wrong

File: OneTrustConsentFilter/lib/build.gradle (hunk 1 above)
Comment (itsdebs) [reply OneTrustConsentFilter/lib/build.gradle]: Till rudderstack sdk is released, this is the one with Consent filter capabilities.

File: OneTrustConsentFilter/gradle.properties (hunk 3 above)
Comment (itsdebs) [reply OneTrustConsentFilter/gradle.properties:31]: Hopefully. We have "BREAKING CHANGE" in commit message.

File: OneTrustConsentFilter/.github/workflows/snapshot_release.yml (hunk 4 above)
Comment (itsdebs) [reply OneTrustConsentFilter/.github/workflows/snapshot_release.yml]: No, these are  organization wide secrets.

**Review by bardisg — APPROVED** [review]

**Review by bardisg — CHANGES_REQUESTED** [review]
ci checks fail. please fix them

**Review by bardisg — APPROVED** [review]

**Pull request merged**